### PR TITLE
feat!: remove `useShareIndexes`

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/1894

This is an API breaking change b/c it changes the function signature of `SplitBlobs`. celestia-node can adopt this change [here](https://github.com/celestiaorg/celestia-node/blob/0823f6b644eb6ba176ef5ae3c008c65018ed5f46/blob/helper.go#L77)

```diff
-	rawShares, err := shares.SplitBlobs(0, nil, b, false)
+	rawShares, err := shares.SplitBlobs(b)
```

### CODE REVIEW

File: pkg/shares/share_splitting.go [hunk 1]
```diff
@@ -72,21 +72,10 @@ func SplitTxs(txs coretypes.Txs) (txShares []Share, pfbShares []Share, shareRang
 	return txShares, pfbShares, mergeMaps(txMap, pfbMap), nil
 }
 
-func SplitBlobs(cursor int, indexes []uint32, blobs []coretypes.Blob, useShareIndexes bool) ([]Share, error) {
-	if useShareIndexes && len(indexes) != len(blobs) {
-		return nil, ErrIncorrectNumberOfIndexes
-	}
+func SplitBlob(blob coretypes.Blob) ([]Share, error) {
```
Comment (cmwaters) [inline pkg/shares/share_splitting.go:75]: Would appreciate if we could start to add go docs for our public functions. Maybe we can go do that in a separate PR

**Review by cmwaters — DISMISSED** [review]
Nice clean up

File: pkg/shares/share_splitting.go (hunk 1 above)
Comment (rootulp) [reply pkg/shares/share_splitting.go:75]: +1 we have https://github.com/celestiaorg/celestia-app/issues/1321 to track existing functions but I'll add godoc to this function b/c added in this PR

File: pkg/shares/share_splitting.go [hunk 2]
```diff
@@ -72,21 +72,22 @@ func SplitTxs(txs coretypes.Txs) (txShares []Share, pfbShares []Share, shareRang
...
+	writer := NewSparseShareSplitter()
+	if err := writer.Write(blob); err != nil {
+		return nil, err
 	}
+	return writer.Export(), nil
+}
+
+// SplitBlobs splits the provided blobs into shares.
+func SplitBlobs(blobs []coretypes.Blob) ([]Share, error) {
```
Comment (Wondertan) [inline pkg/shares/share_splitting.go:85]: What about unifying them into a single `SplitBlobs` with variadic param? 
Looks like a perfect match.

File: pkg/shares/share_splitting.go [hunk 3]
```diff
@@ -72,21 +72,22 @@ func SplitTxs(txs coretypes.Txs) (txShares []Share, pfbShares []Share, shareRang
...
+// SplitBlob splits the provided blob into shares.
+func SplitBlob(blob coretypes.Blob) ([]Share, error) {
+	writer := NewSparseShareSplitter()
+	if err := writer.Write(blob); err != nil {
+		return nil, err
 	}
+	return writer.Export(), nil
+}
+
+// SplitBlobs splits the provided blobs into shares.
```
Comment (cmwaters) [inline pkg/shares/share_splitting.go:84]:
```suggestion
// SplitBlobs splits the provided blobs into shares belonging to the same sequence. All blobs should thus belong to the same namespace
```
I think it's important to specify that they should be of the same namespace. Does the `SparseShareSplitter` already panic if shares are from different namespaces?

Comment (rootulp) [reply pkg/shares/share_splitting.go]: `SplitBlobs` can be called with blobs from different namespaces. SparseShareSplitter supports writing blobs from different namespaces.

Going to revert this comment.

File: pkg/shares/share_splitting.go (hunk 2 above)
Comment (rootulp) [reply pkg/shares/share_splitting.go:85]: Can do! Thanks for the suggestion.

File: pkg/shares/share_splitting.go (hunk 3 above)
Comment (rootulp) [reply pkg/shares/share_splitting.go:84]: Resolved by 245d2431d57eb7ff310ff3814d9fc41cfc9055fc

File: pkg/shares/share_splitting.go (hunk 2 above)
Comment (rootulp) [reply pkg/shares/share_splitting.go]: Resolved by [10f5f55](https://github.com/celestiaorg/celestia-app/pull/1895/commits/10f5f552855e7cb8888776fb6a34b3fbab25513d)

File: pkg/shares/share_splitting.go [hunk 4]
```diff
@@ -72,21 +72,10 @@ func SplitTxs(txs coretypes.Txs) (txShares []Share, pfbShares []Share, shareRang
 	return txShares, pfbShares, mergeMaps(txMap, pfbMap), nil
 }
 
-func SplitBlobs(cursor int, indexes []uint32, blobs []coretypes.Blob, useShareIndexes bool) ([]Share, error) {
```
[the remark this answers is not in the document]
Comment (rootulp) [reply pkg/shares/share_splitting.go:75]: Resolved by [10f5f55](https://github.com/celestiaorg/celestia-app/pull/1895/commits/10f5f552855e7cb8888776fb6a34b3fbab25513d)

**Review by cmwaters — APPROVED** [review]
🧹

**Comment by rootulp** [comment]
https://github.com/celestiaorg/celestia-app/actions/runs/5260800176/jobs/9508250062?pr=1895#step:6:548 fails because https://github.com/celestiaorg/devops/issues/326

**Review by evan-forbes — APPROVED** [review]
nice, very clean

**Comment by Wondertan** [comment]
Is this PR supposed to be in the following rc?

**Comment by rootulp** [comment]
IMO it isn't critical to get into 1.0.0 so it can go in next 1.0.0-rc (if there is one) or in a subsequent release.

**Comment by cmwaters** [comment]
I think this should just be part of v1.1

**Review by cmwaters — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]
